overlord/overlord.go

```diff
@@ -93,6 +101,34 @@ func loadState(backend state.Backend) (*state.State, error) {
 	return state.ReadState(backend, r)
 }
 
+// Run runs a loop to ensure the current state regularly through StateEngine Ensure().
+func (o *Overlord) Run() {
+	intv := ensureInterval
+	o.loopTomb.Go(func() error {
+		timer := time.NewTimer(intv)
+		for {
+			select {
+			case <-o.loopTomb.Dying():
+				return nil
+			case <-timer.C:
+			}
+			err := o.stateEng.Ensure()
+			if err != nil {
+				logger.Noticef("state engine ensure failed: %v", err)
+				// continue to the next Ensure() try for now
+			}
+			timer.Reset(intv)
+		}
+	})
+}
+
+// Stop stops the ensure loop and the managers under the StateEngine.
+func (o *Overlord) Stop() error {
+	o.loopTomb.Kill(nil)
+	o.loopTomb.Wait()
```

Comment:
We shouldn't drop the error here.. I suggest this phrasing, for clarity:
```
err1 := o.loopTomb.Wait()
err2 := o.stateEng.Stop()
if err1 != nil {
        return err1
}
return err2
```

I think it makes sense to give precedence to the overlord error, if it exists, as it has a global notion of the system. But we may change our minds about that as we learn more.